compiler/ztests/join.yaml

```diff
@@ -0,0 +1,15 @@
+script: |
+  super compile -dag -C 'join as {j,k} on k.kkey=j.jkey'
+  ! super compile -dag -C 'join as {j,j} on key'
+
+outputs:
+  - name: stdout
+    data: |
+      null
+      | inner join as {j,k} on jkey=kkey
+      | output main
+  - name: stderr
+    data: |
+      left and right join aliases cannot be the same at line 1, column 6:
+      join as {j,j} on key
+           ~~~~~~~~
```

Comment:
Nit: I don't think the span should include "as" on these errors.

Reply:
It's easiest to do it the way I did but I can see if I can rewrite the peg rules to allow for this.

Reply:
Just a nit so your call.